composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java

```diff
@@ -146,91 +145,116 @@ public void run() {
     }
 
     public void streamOutput() {
-        BufferedReader reader = null;
-        try {
-            reader = new BufferedReader(new InputStreamReader(this.command.getProgram().getInputStream(), Charset
-                    .defaultCharset()));
-            String line = "";
-            while ((line = reader.readLine()) != null) {
-                // improve "server connector started" log message to have the service URL in it.
-                // This is to handle the cloud use case.
-                if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD)
-                        && getServerStartedURL() != null) {
-                    this.updatePort(getServerStartedURL());
-                    line = LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD + " "
-                            + getServerStartedURL();
-                }
-
-                // This is to handle local service run use case.
-                if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_LOCAL)
-                        && getServerStartedURL() == null) {
-                    this.updatePort(line);
-                    pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
-                } else {
-                    pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
+        try (InputStream inputStream = this.command.getProgram().getInputStream()) {
+            while (this.command != null && this.command.getProgram().isAlive()) {
+                String output = consumeStreamOutput(inputStream);
+                if (!output.isEmpty()) {
+                    String[] lines = output.split(System.lineSeparator());
+                    for (String line : lines) {
+                        // improve "server connector started" log message to have the service URL in it.
+                        // This is to handle the cloud use case.
+                        if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD)
+                                && getServerStartedURL() != null) {
+                            this.updatePort(getServerStartedURL());
+                            line = LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD + " "
+                                    + getServerStartedURL();
+                        }
+
+                        // This is to handle local service run use case.
+                        if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_LOCAL)
```

Comment:
Shall we simplify the If-Else Clause by taking out the pushToClient method call out?

Reply:
I was reluctant to change the existing logic as the second `If` logic has a dependency on the first `If` logic(aka. `line` variable)